plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java

```diff
@@ -352,6 +352,9 @@ private boolean validateMavenSettings() {
     } else if (!MavenCoordinatesValidator.validateArtifactId(artifactId)) {
       setErrorMessage(Messages.getString("ILLEGAL_ARTIFACT_ID", artifactId)); //$NON-NLS-1$
       return false;
+    } else if (ResourcesPlugin.getWorkspace().getRoot().getProject(artifactId).exists()) {
+      setErrorMessage(artifactId + " project already exists in this workspace"); //$NON-NLS-1$
```

Comment:
Oh, another one: our UI doesn't hint that the artifact ID will become the project name, so saying "<some name> project already exists" may be confusing, as it might imply that a "project" (as an abstract notion) already exists, while it is actually the project _name_ that already exists.